content/tutorials/reactjs-realtime-commenting.textile

```diff
@@ -343,41 +370,36 @@ class App extends Component {
 }
 
 export default App;
-```
+div.
 
 This contains the components we created earlier. We define a @comments@ state which is an array of comments. It defaults to an empty array. This will be updated as we add comment through the comment form. Also, we pass the @comments@ state as props to the Comments component. This is how the Comments component get the comments it renders.
 
-"See this step in Github":https://github.com/ably/tutorials/tree/reactjs-commenting-step12
+"See this step in Github":https://github.com/marsielko/tutorials/tree/aadcf9f06d1e5bd69b0d4063d51cbc5b364c6e98
 
 h2. Step 13 - Updating State With Newly Added Comment
 
-Recall from step 9 above, we called @handleAddComment()@ which is responsible for adding a new comment to @state@. But we are yet to create this function, let's do that now. Within @components/App.js@, add the code below just before @render()@:
+Within @src/components/App.js@, add the following code just before @render()@:
 
-```[javascript]
+bc..
 handleAddComment(comment) {
   this.setState(prevState => {
     return {
-      comments: prevState.comments.concat(comment)
+      comments: [comment].concat(prevState.comments)
     };
   });
 }
-```
+div.
 
 The code above adds the comment that was made to @state@. This way, the comments list is updated with new comments in realtime.
 
-Next, still within @components/App.js@, let’s bind @handleAddComment()@ to the @this@ keyword by adding this line to the @constructor()@ just after @super()@:
+Next, still within @src/components/App.js@, let’s bind @handleAddComment()@ to the @this@ keyword by adding this line to the @constructor()@ just after @super()@:
 
-```[javascript]
+bc..
 this.handleAddComment = this.handleAddComment.bind(this);
-```
-
-Also, still within @components/App.js@, pass @handleAddComment@ as props to @CommentBox@ component as below:
+div.
 
-```[javascript]
-<CommentBox handleAddComment={this.handleAddComment} />
-```
 
-"See this step in Github":https://github.com/ably/tutorials/tree/reactjs-commenting-step13
+"See this step in Github":https://github.com/marsielko/tutorials/tree/f3e0e74969c08ccbee54cd36c20d8ac557426ec9
```

Comment:
is this Github repo correct?

Reply:
No, but at that time I couldn't tell what the exact URL will be, so i assumed that this URL will be updated at some point. I just used one that works.

Reply:
@marsielko What's required to be done here?

Reply:
@tbedford The working URL to given step in the tutorial. I also updated the examle code of a tutorial, which isn't merged yet and I wasn't sure whether the used URL will work, so I changed the URL to point to my fork. When the example code for tutorial is merged, this URL can be updated.

Reply:
@tomczoink - your advice please? I feel these links should be directly to Ably. How do you normally deal with this issue? We don't want to have to create another PR and review again to update links.

Reply:
@marsielko: OK I think I understand now, you have another PR on the ably/tutorials repo for the tutorial code, and you are waiting for that PR to be approved and merged, before updating the links in this PR. That seems to make sense!